gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/process/traversal/step/map/AsDateStepTest.java

```diff
@@ -45,13 +46,12 @@ protected List<Traversal> getTraversals() {
     @Test
     public void shouldParseDate() {
 
-        final Instant testInstant = ZonedDateTime.of(2023, 8, 2, 0, 0, 0, 0, UTC).toInstant();
-        final Date testDate = new Date(testInstant.getEpochSecond() * 1000);
+        final OffsetDateTime testDate = OffsetDateTime.of(LocalDateTime.of(2023, 8, 2, 0, 0, 0, 0), UTC);
 
-        assertEquals(new Date(1), __.__(1).asDate().next());
-        assertEquals(new Date(3), __.__(3L).asDate().next());
-        assertEquals(new Date(6), __.__(new BigInteger("6")).asDate().next());
-        assertEquals(testDate, __.__(testDate.getTime()).asDate().next());
```

Comment:
You should add tests where Date is the input traverser. Same goes for the DateAddStepTest and DateDiffStepTest

Reply:
Yup I can do that for asDate and dateAdd (tests in dateDiff were added already)